### ISSUE TITLE
Error in performing DA Analysis

### ISSUE THREAD

**Issue by SrijaniBagchi** [ELELAB/CAncer-bioMarker-Prediction-Pipeline-CAMPP#3]
I want to obtain differentially expressed genes over samples of patients infected with a disease along with control samples. I have used the following command:
Rscript CAMPP.R -d data2.xlsx -m metadata2.xlsx -v array -g Accession,Code -t log2 -l 0.0 -s TRUE -f 0.6,0.05 -n DEG
However, this returns the following error:
Error: Aesthetics must be either length 1 or the same as the data (32): fill and colour
Backtrace:
     █
  1. └─ggplot2::ggsave(...)
  2.   ├─grid::grid.draw(plot)
  3.   └─ggplot2:::grid.draw.ggplot(plot)
  4.     ├─base::print(x)
  5.     └─ggplot2:::print.ggplot(x)
  6.       ├─ggplot2::ggplot_build(x)
  7.       └─ggplot2:::ggplot_build.ggplot(x)
  8.         └─ggplot2:::by_layer(function(l, d) l$compute_aesthetics(d, plot))
  9.           └─ggplot2:::f(l = layers[[i]], d = data[[i]])
 10.             └─l$compute_aesthetics(d, plot)
 11.               └─ggplot2:::f(..., self = self)
 12.                 └─ggplot2:::check_aesthetics(evaled, n)
Execution halted

Do I need some modifications in my data for this command to work?

**Comment by ThildeBT** [comment]
Hi @SrijaniBagchi :-) 

Okay so from what you write, I have a couple of ideas as to what could be the issue:

1. Remove the flag **-l**, as it specifies lasso regression. Even if you do want that to perform this analysis in addition to your DEA, the parameter cannot be set to 0.0 (it should be between 0.1-1.0), as this is essentially ridge regression, which CAMPP does not support. This is stated somewhere in the way to long manual I hope ;-) So this instead:

**Rscript CAMPP.R -d data2.xlsx -m metadata2.xlsx -v array -g Accession,Code -t log2 -s TRUE -f 0.6,0.05 -n DEG**

2. The name of your **id** variable in the above is = **Accession**, I am just checking that you are giving it a sample ID and not a gene/protein id? 
Also the framework which makes the DEA contrasts and feeds it to limma, does not do well with groups (here your **Code**) which consists only of integers/numbers. So, if this is the case you should add a letter to the group names to make sure that it is understood as a string. Also, the group names cannot have a **-** in them, as this character is used as the symbol for contrast in the design matrix.
Make sure that **Code** has atleast two groups with 5 samples in each group. 

3.  Make sure your matrices both _data_ en _metadata_ look like the files found here: https://github.com/ELELAB/N-glycan-TIF/tree/master/Data/DataExamples. You can see an example of how the data need to be set up, you can even download and test the pipeline with these data.

4. Check that your R version in >= 4.0.0 

If non of the suggestions above fix your problem, let me know, and I will have a deeper look into it when I have the time.

Cheers,

Thilde

**Comment by SrijaniBagchi** [comment]
Hey @ThildeBT , 
I have followed your instructions but the same error persists. Here is what my datafiles look like:

I am using R-version 4.0.0(2020-04-24) and altered by command to:
Rscript CAMPP.R -d data2.xlsx -m metadata2.xlsx -v array -g Accession,Code -t log2 -s TRUE -f 0.6,0.05 -n DEG

It would be a great help if you could take a look at it.
Thanks and Regards,
Srijani

**Comment by ThildeBT** [comment]
Hey @SrijaniBagchi,

Okay I have figured it out. You had the sample ids with all capital letters in the data file and with a small **o** in the metadatafile. They need to match. However I have now corrected the code so the paring is not case dependent as this is of course more robust. I have also corrected the code a little so in another way, so i suggest you to re-download the repository or rater the scripts.

Here are your datafiles back, corrected also for the **/** in the gene names as this may be a problem you would otherwise run into.

As soon as you have downloaded them I suggest you to remove them from your comment above, and I will do the same, so they are not visible here to others :-) 

On a different note, I noticed that one of your sample appears to be an exteme outlier from the MDS plot. 

Cheers,

Thilde
